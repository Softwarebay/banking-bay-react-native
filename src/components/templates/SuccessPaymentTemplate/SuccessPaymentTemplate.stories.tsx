import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { CenterStory } from 'utils/storybook/CenterStory';

import { SuccessPaymentTemplate } from './SuccessPaymentTemplate';

storiesOf('Templates/SuccessPaymentTemplate', module)
	.addDecorator((getStory) => (
		<CenterStory style={{ paddingHorizontal: 0 }}>{getStory()}</CenterStory>
	))
	.add('Default', () => (
		<SuccessPaymentTemplate
			amount="10.57"
			primaryButtonTitle={text('primaryButtonTitle', 'Primary')}
			secondaryButtonTitle={text('secondaryButtonTitle', 'Secondary')}
			onPressPrimaryButton={action('onPressPrimaryButton')}
			onPressSecondaryButton={action('onPressSecondaryButton')}
		/>
	));
