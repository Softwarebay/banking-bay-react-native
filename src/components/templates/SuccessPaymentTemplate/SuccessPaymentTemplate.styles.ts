import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	amount: {
		marginVertical: 20,
	},
	amountFloat: {
		fontSize: 20,
	},
	amountInt: {
		fontSize: 33,
	},
	amountText: {
		fontFamily: 'MulishBold',
	},
	backgroundImg: {
		flex: 1,
	},
	center: {
		alignItems: 'center',
		flex: 1,
		justifyContent: 'center',
	},
	container: {
		flex: 1,
		paddingBottom: 30,
		paddingHorizontal: 20,
	},
	secondaryBtn: {
		marginBottom: 14,
	},
	successIcon: {
		width: 200,
	},
	successMessage: {
		marginHorizontal: 62,
		textAlign: 'center',
	},
});
