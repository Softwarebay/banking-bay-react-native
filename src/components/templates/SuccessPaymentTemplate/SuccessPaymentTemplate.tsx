import { ImageBackground, View } from 'react-native';
import { StatusBar } from 'expo-status-bar';

import { Button, FloatNum, Heading, Image, Text } from 'components/atoms';

import { bgSummary } from 'assets/backgrounds';
import { colors } from 'theme';

import { styles } from './SuccessPaymentTemplate.styles';

interface Props {
	amount: string;
	primaryButtonTitle: string;
	secondaryButtonTitle: string;
	onPressPrimaryButton: () => void;
	onPressSecondaryButton: () => void;
}

export const SuccessPaymentTemplate = ({
	amount,
	primaryButtonTitle,
	secondaryButtonTitle,
	onPressPrimaryButton,
	onPressSecondaryButton,
}: Props) => {
	return (
		<>
			<StatusBar style="dark" />
			<ImageBackground source={bgSummary} style={styles.backgroundImg}>
				<View style={styles.container}>
					<View style={styles.center}>
						<Image name="success-icon" style={styles.successIcon} />
						<Heading
							color={colors.green}
							size="h2"
							testID="payment-success-heading-test-id"
							text="Success!"
						/>
						<FloatNum
							floatStyle={[styles.amountText, styles.amountFloat]}
							intStyle={[styles.amountText, styles.amountInt]}
							number={amount}
							prefix="$ "
							style={styles.amount}
						/>
						<Text
							style={styles.successMessage}
							testID="payment-success-message-test-id"
							text="Your payment has been processed!"
						/>
					</View>
					<Button
						style={styles.secondaryBtn}
						testID="payment-success-secondary-btn-test-id"
						title={secondaryButtonTitle}
						type="secondary"
						onPress={onPressSecondaryButton}
					/>
					<Button
						testID="payment-success-primary-btn-test-id"
						title={primaryButtonTitle}
						onPress={onPressPrimaryButton}
					/>
				</View>
			</ImageBackground>
		</>
	);
};
