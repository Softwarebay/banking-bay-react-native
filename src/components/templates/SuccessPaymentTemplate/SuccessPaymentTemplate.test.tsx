import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { SuccessPaymentTemplate } from './SuccessPaymentTemplate';

import type { Props } from './SuccessPaymentTemplate';

describe('SuccessPaymentTemplate', () => {
	const onPressPrimaryButton = jest.fn<Props['onPressPrimaryButton'], []>();
	const onPressSecondaryButton = jest.fn<Props['onPressSecondaryButton'], []>();

	const props: Props = {
		amount: '10.00',
		primaryButtonTitle: 'Primary Action',
		secondaryButtonTitle: 'Secondary Action',
		onPressPrimaryButton,
		onPressSecondaryButton,
	};

	it('should match snapshot', () => {
		const tree = testRenderer
			.create(<SuccessPaymentTemplate {...props} />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render success image', () => {
		render(<SuccessPaymentTemplate {...props} />);

		const img = screen.getByTestId('success-icon-img-test-id');

		expect(img).not.toBeNull();
	});

	it('should render success heading', () => {
		render(<SuccessPaymentTemplate {...props} />);

		const heading = screen.getByTestId('payment-success-heading-test-id');

		expect(heading).toHaveTextContent('Success!');
	});

	it('should render success message', () => {
		render(<SuccessPaymentTemplate {...props} />);

		const message = screen.getByTestId('payment-success-message-test-id');

		expect(message).toHaveTextContent('Your payment has been processed!');
	});

	it('should display amount of transferred money', () => {
		render(<SuccessPaymentTemplate {...props} />);

		const transferValue = screen.getByTestId(
			'payment-success-float-num-test-id'
		);

		expect(transferValue).toHaveTextContent('$ 10.00');
	});

	it('should display primary and secondary button', () => {
		render(<SuccessPaymentTemplate {...props} />);

		const primaryBtn = screen.getByTestId(
			'payment-success-primary-btn-test-id'
		);

		const secondaryBtn = screen.getByTestId(
			'payment-success-secondary-btn-test-id'
		);

		expect(primaryBtn).toHaveTextContent('Primary Action');
		expect(secondaryBtn).toHaveTextContent('Secondary Action');
	});

	it('should handle press events on primary button', () => {
		render(<SuccessPaymentTemplate {...props} />);

		const primaryBtn = screen.getByTestId(
			'payment-success-primary-btn-test-id'
		);

		fireEvent.press(primaryBtn);

		expect(onPressPrimaryButton).toHaveBeenCalledTimes(1);
		expect(onPressSecondaryButton).not.toHaveBeenCalled();
	});

	it('should handle press events on secondary button', () => {
		render(<SuccessPaymentTemplate {...props} />);

		const secondaryBtn = screen.getByTestId(
			'payment-success-secondary-btn-test-id'
		);

		fireEvent.press(secondaryBtn);

		expect(onPressSecondaryButton).toHaveBeenCalledTimes(1);
		expect(onPressPrimaryButton).not.toHaveBeenCalled();
	});
});
