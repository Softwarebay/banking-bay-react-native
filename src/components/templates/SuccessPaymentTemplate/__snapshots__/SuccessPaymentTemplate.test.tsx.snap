// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessPaymentTemplate should match snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "paddingBottom": 30,
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": undefined,
            },
            Object {
              "aspectRatio": 1.0749464668094217,
            },
            Object {
              "width": 200,
            },
          ]
        }
        testID="success-icon-img-test-id"
      />
      <Text
        style={
          Array [
            Object {
              "color": "#3EB290",
            },
            Object {
              "fontFamily": "MulishBold",
              "fontSize": 44,
              "lineHeight": 48.4,
            },
            undefined,
          ]
        }
        testID="payment-success-heading-test-id"
      >
        Success!
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "row",
            },
            Array [
              Object {
                "alignItems": "baseline",
              },
              Object {
                "marginVertical": 20,
              },
            ],
          ]
        }
        testID="payment-success-float-num-test-id"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Array [
                Object {
                  "fontFamily": "MulishBold",
                },
                Object {
                  "fontSize": 33,
                },
              ],
            ]
          }
        >
          $ 10.
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Array [
                Object {
                  "fontFamily": "MulishBold",
                },
                Object {
                  "fontSize": 20,
                },
              ],
            ]
          }
        >
          00
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 17,
              "lineHeight": 27.2,
            },
            Object {
              "marginHorizontal": 62,
              "textAlign": "center",
            },
          ]
        }
        testID="payment-success-message-test-id"
      >
        Your payment has been processed!
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#F1F5FD",
            "borderColor": "#CED6E1",
            "borderWidth": 1,
          },
          Object {
            "marginBottom": 14,
          },
        ]
      }
      testID="payment-success-secondary-btn-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "MulishSemiBold",
              "fontSize": 17,
            },
            Object {
              "color": "#1B1D4D",
            },
          ]
        }
      >
        Secondary Action
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          undefined,
        ]
      }
      testID="payment-success-primary-btn-test-id"
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4288142079,
              4279654301,
            ],
            "endPoint": Array [
              1,
              1,
            ],
            "locations": undefined,
            "startPoint": Array [
              0,
              0,
            ],
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "MulishSemiBold",
                "fontSize": 17,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          Primary Action
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</View>
`;
